src/server/db_slice.cc

```diff
@@ -852,6 +852,21 @@ uint64_t DbSlice::RegisterOnChange(ChangeCallback cb) {
   return ver;
 }
 
+void DbSlice::CallChangeOnAllLessThanVersion(DbIndex db_ind, PrimeIterator it, uint64_t version) {
+  uint64_t bucket_version = it.GetVersion();
+  // change_cb_ is ordered by version.
+  for (const auto& ccb : change_cb_) {
```

Comment:
I think there is a bug here if `ccb.second` preempts and some other snapshot (un)registers to `change_cb_`.
changing `change_cb_` type to map might fix the issue.

Reply:
vesion -> version

Reply:
Our callbacks do not preempt. We should not allow for preemptable callbacks as they are getting iterator and it could be invalidated if there is a preempt.

Reply:
our callbacks can preempt when writing to the channel

Reply:
@dranikpg you are right. They can preempt when pushing to channel. In this case we should check if we dont have a bug in the flow, as we are using iterator..